src/pages/audio-rewards-page/hooks.ts

```diff
@@ -107,3 +101,41 @@ export const useOptimisticUserChallenge = (
     state: getUserChallengeState(challengeOverridden)
   }
 }
+
+/**
+ * Get all user challenges but in client-side optimistic state
+ * @see useOptimisticUserChallenge
+ * @returns all user challenges
+ */
+export const useOptimisticUserChallenges = (): Partial<
+  Record<ChallengeRewardID, OptimisticUserChallenge>
```

Comment:
nit: probably don't need this type signature since we type the reduce call (could be wrong)